distribution/docs/src/main/resources/content/_managing/_configuring/configuring-multi-factor-auth.adoc

```diff
@@ -0,0 +1,24 @@
+:title: Configuring Multi Factor Authentication
+:type: subConfiguration
+:status: published
+:parent: Configuring REST Services for Users
+:summary: Configuring Multi Factor Authentication using Keycloak as an IdP.
+:order: 23
+
+// This is for MFA
+== {title}
+
+Mutli-factor authentication, sometimes referred to as two-factor authentication, allows for greater security.
+It does this by requiring users to provide multiple proofs of identity, typically through something they know, and something they have/are.
+The IdP that comes with ${branding} does not support multi-factor authentication by default.
+
+Keycloak can be used to help setup and configure multi-factor authentication.
+See <<{managing-prefix}connecting_to_an_external_identity_provider,Connecting to an External Identity Provider>> on how to initially hookup Keycloak.
+
+.Configuring Keycloak for MFA
+. Download and install Keycloak from here: https://www.keycloak.org/downloads.html[Keycloak Downloads^] {external link}
+. Choose an operation mode and configure it based off this information here: https://www.keycloak.org/docs/latest/server_installation/index.html#_operating-mode[Server Installation^] {external link}
+. Setup an Admin User following these steps here: https://www.keycloak.org/docs/latest/server_admin/index.html#server-initialization[Setup Server Admin^] {external link}
+. Refer to https://www.keycloak.org/docs/latest/server_admin/index.html#otp-policies[OTP Policies^] {external link} for how to setup multi-factor authentication using supported authentication tools such as *FreeOTP* and *Google Authenticator*.
+
+See the https://www.keycloak.org/documentation.html[Keycloak documentation^] {external link} for more information and details about how to configure Keycloack for multi-factor authentication.
```

Comment:
How does Keycloak integrate with DDF or its Idp? Is there something in DDF that has to be configured?

Reply:
This is addressed by pointing to the `Connecting to an External Identity Provider` docs section on line 16 on how to hookup Keycloak. This should suffice for the time being because Keycloak is being installed and ran externally as well at the moment.